TLM/TLM/Util/Record/LaneArrowsRecord.cs

```diff
@@ -29,13 +38,15 @@ public class LaneArrowsRecord : IRecordable {
                 LaneArrowManager.LANE_TYPES,
                 LaneArrowManager.VEHICLE_TYPES,
                 sort: false);
-            foreach(var lane in lanes) {
+            foreach (var lane in lanes) {
                 LaneArrowsRecord laneData = new LaneArrowsRecord {
                     LaneId = lane.laneId,
                 };
                 ret.Add(laneData);
```

Comment:
If we know how many lanes there are (`lanes` is an array or list I assume?) then we can make a small optimisation by initialising `ret` list with that capacity.

```cs
var ret = new List<LaneArrowsRecord>(lanes.Length); // or .Count if a list
```

It will remove the need to constantly adjust the size of the `ret` list while it's being populated, resulting in less memalloc calls.

Also, if there are lots of lists, it would be worth checking the capacity vs. count values. If there is excess capacity it is sometimes beneficial to use [`.TrimExcess()`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.trimexcess?view=netcore-3.1) to remove excess capacity.

Reply:
There is a lot of optimisation potential. But it just does not matter. we are not at performance critical part of the code.

Reply:
Also egi told me allocation is fast with .net framwork

Reply:
Initializing a List<T> with a good capacity value (if you know it will be > 4) is less about the allocations and more about the array copying.
Default capacity for a List<T> is 4.
If you know you will have around 20 items, why start with 4, copy array content to 8, to 16, to 32?
That is just wasteful, for memory and cpu cycles.
Its true that this optimization won't matter in the end in non perf critical code, but it will still matter a little and it could add up.
The way I see it is, you should always consider performance, but there is a tradeoff to be done with other code qualities.
Code readability would be such a quality to weight against.
In non perf critical code I would always go for code readability > performance.
But initializing a List<T> with a better capacity value does not reduce the readability, so you don't lose anything (ok in this case you lose some time).
If you make it a habit to always consider List<T> capacity, its worth the time I'd say, so I consider it good practice.
Of course you should only do that if you know for sure that the content will exceed the initial capacity of 4.
And if the number of items is fixed anyway, I don't see a reason to use List<T> in the first place and would go for Array<T>.

Reply:
This is not part of the code I modified and we don't know the number of lanes in advance, but I know maximum number of lanes.

Reply:
How often is the code being invoked? If only very rarely then not too much problem.

Reply:
I already changed it. it's only happens when user undo-redo stuff. or copy paste based on move it.

Reply:
a far more important issue is the binary serialization method I am using. @krzychu124 and @kvakvs say I should use something more flexible to version change.